### ISSUE TITLE
Move "loading baselines" behind a button

### ISSUE THREAD

**Issue by FransdeJong** [umbraco/Umbraco.Cloud.Issues#367]
When starting a new Cloud project you always have to wait for the possible baselines to load.
This is unnecessary when you want to start without a baseline.

The more projects you have the slower this process seems to get.

Instead of prepopulating the dropdown it would be less of a wait to populate the dropdown when expanding or use a button and a modal to select the baseline.

**Comment by sitereactor** [comment]
I agree with this, and we have had similar discussions internally ad well. Unfortunately not something that has materialized as of yet.

In my opinion the creation of a Baseline project should be moved away from the regular Create Project dialog and under the Project page similar to how its possible to add/remove environments. I see the flow as being very similar to creating a new repository on GitHub versus Forking an existing repository on GitHub.
When creating a new Baseline you'd then go to the Project you want to create it from (the Master), click a button, enter a name for the new Project and select a tier and then you are good to go.

**Comment by FransdeJong** [comment]
That sounds like a more logical flow. If I want to update a child project I already have to go to the parent project so all child project management would be in the same place.

One of the requirements for this flow would be a better dashboard that shows the relations between projects. Otherwise it's way to easy to loose track on what is a parent and what is a child.

**Comment by sitereactor** [comment]
Yes, the project overview page could use some love 🙂

**Comment by sajumb** [comment]
Just a quick update. We are currently handling this feature request about only loading baselines when they are actually needed. We expect this improvement to be put into S1 within the next couple of weeks.

**Comment by sitereactor** [comment]
I have another perspective on this.
The Baseline creation should be moved to the Project Page to make it super simple and quick to create a Baseline child site.
This feature has always been inspired by GitHub repositories and Forks. In order to create a Fork (Baseline child) you go to the  GitHub project you want to Fork and click the Fork button. You now get a Fork of the original GitHub repository under your own account.
Similarly, it would make sense that any project has a button, which enables you to create a Baseline child project based on the Umbraco Cloud Project you are currently in. The only you would then need to select is the Cloud Plan and the name for the Project.
Quick and easy + it would allow us to remove Baselines from the regular Create Project dialogue.
(If you have access to a lot of projects you'd have a very long list any way - not particularly nice to scroll through regardless of how its loaded). // @sajumb

**Comment by sitereactor** [comment]
(i realize that I'm kinda repeating myself here, but I'm unsure if you read and considered previous comments)

**Comment by sajumb** [comment]
Hi @sitereactor
Your and @FransdeJong thoughts and idea sharing have been noted and very much appreciated. We definitely want to introduce a more simple and quick creation of a new project from an existing project page. It is quite up there in our product backlog and we have planned to sketch the first prototype next week.

As you state we still need the user to select a plan, name, owner, technical contacts and possible more in the future. So it cannot be a standalone one-click feature. Currently, I rather see option from a project page as a shortcut to step 2 of the 3 step wizard, with the baseline pre-filled in step 3. I will contact you and Frans next week, so we can outline the best approach for this new feature.

The request by Frans is quite easy for us to implement. And as we believe the list of potential baselines still will have to be shown for at least one future path in creation a project based on a baseline, we have decided that this performance improvement finally should be prioritized. And actually kick off the overall improved baseline initiative.
